c_tests/unit.c

```
@@ -610,29 +637,63 @@ convert_image_yuv_to_rgb_size_mode_stride(uint32_t       num_planes,
         }
     }
 
-    for (y = 0; y < chroma_height; y++, input += chroma_fill_bytes) {
-        uint32_t x;
+    if (format == DCP_PIXEL_FORMAT_NV12) {
+        for (y = 0; y < chroma_height; y++, input += u_chroma_fill_bytes) {
+            uint32_t x;
 
-        for (x = 0; x < width; x += 2, input += 2) {
-            uint32_t index = (x >> 1) & 0x7;
-            input[0] = cb_to_rgb_input[color_space_index][index];
-            input[1] = cr_to_rgb_input[color_space_index][index];
+            for (x = 0; x < width; x += 2, input += 2) {
+                uint32_t index = (x >> 1) & 0x7;
+                input[0] = cb_to_rgb_input[color_space_index][index];
+                input[1] = cr_to_rgb_input[color_space_index][index];
+            }
+        }
+    } else if (format == DCP_PIXEL_FORMAT_I420) {
+        for (y = 0; y < chroma_height; y++, input += u_chroma_fill_bytes) {
+            uint32_t x;
+
+            for (x = 0; x < width; x += 2, input++) {
+                uint32_t index = (x >> 1) & 0x7;
+                input[0] = cb_to_rgb_input[color_space_index][index];
+            }
+        }
+
+        for (y = 0; y < chroma_height; y++, input += v_chroma_fill_bytes) {
+            uint32_t x;
+
+            for (x = 0; x < width; x += 2, input++) {
+                uint32_t index = (x >> 1) & 0x7;
+                input[0] = cr_to_rgb_input[color_space_index][index];
+            }
         }
+    } else {
+        return;
```

I would return a failure there.